packages/config/src/ProfileInfo.ts

```diff
@@ -920,10 +920,8 @@ export class ProfileInfo {
                 if (argNameFound || missingArgsIndex >= 0) {
                     if (knownArgsIndex < 0) {
                         mergedArgs.knownArgs.push(tempArg);
-                    } else {
-                        mergedArgs.knownArgs[knownArgsIndex] = tempArg;
                     }
-                    if (missingArgsIndex >= 0) delete mergedArgs.missingArgs[missingArgsIndex];
+                    if (missingArgsIndex >= 0) mergedArgs.missingArgs = mergedArgs.missingArgs.splice(missingArgsIndex, 1);
```

Comment:
Should the left side of equal sign be removed?
`mergedArgs.missingArgs = mergedArgs.missingArgs.splice(missingArgsIndex, 1);` -> `mergedArgs.missingArgs.splice(missingArgsIndex, 1);`

I believe `Array.splice` returns the deleted item(s), not the remaining item(s) in the array.

Reply:
`/facepalm` 🤦

Reply:
I believe this is fixed, and enhanced the test so that this behavior is properly captured.